### PR TITLE
Support for custom isItemSelectable function

### PR DESCRIPTION
**Pull request by artem-dudarev**
This will allow to provide a function that will determine whether the catalog item can be selected

### CODE REVIEW

**Comment by artem-dudarev** [comment]
Oops, sorry

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/common/catalog-items-select.js [hunk 1]
```diff
@@ -108,12 +108,12 @@
...
 
         var newBlade = {
-            id: 'CatalogItemsSelect',
+            id: blade.id,
             breadcrumbs: blade.breadcrumbs,
             catalogId: blade.catalogId,
             catalog: blade.catalog,
-            controller: 'virtoCommerce.catalogModule.catalogItemSelectController',
-            template: 'Modules/$(VirtoCommerce.Catalog)/Scripts/blades/common/catalog-items-select.tpl.html',
+            controller: blade.controller,
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/common/catalog-items-select.js]: Are you sure what these changes won't affect other places where this control used?

Comment (artem-dudarev) [reply VirtoCommerce.CatalogModule.Web/Scripts/blades/common/catalog-items-select.js]: I'm pretty sure it won't affect anything, but I can remove it, if you wish.

Comment (eMazeika) [reply VirtoCommerce.CatalogModule.Web/Scripts/blades/common/catalog-items-select.js]: looks like a nice code clean-up, just double-check to be sure

**Pull request merged**